docs2/site/docs/getting-started/errors.md

```diff
@@ -1,78 +1,149 @@
 # Error Handling
 
-The `ExecutionResult` provides an `Errors` property which includes any errors encountered
-during execution. Errors are returned [according to the spec](https://graphql.github.io/graphql-spec/June2018/#sec-Errors),
-which means stack traces are excluded. The `ExecutionResult` is transformed to what the spec
-requires using one or the another `IDocumentWriter`.
+Errors within the GraphQL engine can be thought of as falling into one of three groups: schema errors, input errors, and processing errors.  For instance, if an invalid query was passed to the `DocumentExecuter`, it would be considered an input error, and a `SyntaxError` would be thrown. Or if an invalid enum string was passed as a variable to a query, an `InvalidValueError` would be thrown. All validation rules that fail their respective tests are treated as input errors.  Processing errors typically represent unanticipated exceptions occurring during the execution of a field resolver, such as a timeout during a database operation.
 
-For example `GraphQL.NewtonsoftJson.DocumentWriter` uses [JSON.NET](https://www.nuget.org/packages/Newtonsoft.Json)
-and `GraphQL.SystemTextJson.DocumentWriter` uses new .NET Core memory optimized serializer from
-[`System.Text.Json`](https://docs.microsoft.com/en-us/dotnet/api/system.text.json). For JSON.NET you can change
-what information is provided by setting your values (`ContractResolver`, Converters`, etc.)
-in `JsonSerializerSettings` passed to `DocumentWriter` constructor. For `System.Text.Json`
-serializer you can configure `JsonSerializerOptions` passed to `DocumentWriter` constructor.
+Input errors and processing errors are returned from the `DocumentExecuter` within the `ExecutionResult.Errors` property as a list of `ExecutionError` objects.  `ExecutionError` inherits from `Exception`, and the `Message` property is serialized according to the spec with location and path information.  In addition, three additional pieces of information are serialized to the `extensions` property of the GraphQL error which contain:
+* Within `code`, the `Code` property of the `ExecutionError`, if any,
+* Within `codes`, the `Code` property of the `ExecutionError` along with generated codes of any inner exceptions, if any, and
+* Within `data`, the contents of the `ExecutionError.Data` property, which by default contains the data of the inner exception, if any.
 
-You can also implement your own `IDocumentWriter`.
+Note that by default, messages from unhandled processing errors are masked and a generic "Error trying to resolve product" or similar error is returned.
 
-To help debug errors, you can set `ExposeExceptions` on `ExecutionOptions` which will expose error stack traces.
+Here is a sample result of a `FormatException` thrown within a `product` field resolver (a processing error):
+```json
+{
+  "errors": [
+    {
+      "message": "Error trying to resolve product.",
```

Comment:
```suggestion
      "message": "Error trying to resolve field 'product'.",
```

Reply:
lol - copy and paste from my production code, which is a bit out of date with our new changes!